src/PSRule.Rules.Azure/rules/Azure.Common.Rule.ps1

```diff
@@ -178,15 +178,47 @@ function global:SupportsHybridUse {
     }
 }
 
-function global:IsLinuxOS {
+function global:IsLinuxOS { # TODO doesn't need to be global
+    [CmdletBinding()]
```

Comment:
This function doesn't need to be global but I didn't manage to make it work by declaring it as local

Reply:
The discovery and caching process of rules requires a global scoping. Which is a little different then how a PowerShell script by itself would operate. Functions that are in a local or script scope are not available when the rules execute.

See [Define helper functions](https://microsoft.github.io/PSRule/v2/scenarios/azure-resources/azure-resources/#define-helper-functions).